packages/asset-swapper/src/utils/market_operation_utils/path_optimizer.ts

```diff
@@ -233,14 +253,21 @@ function findRoutesAndCreateOptimalPath(
                 opts.outputAmountPerEth,
                 opts.inputAmountPerEth,
                 fees,
-            )[0];
-            // NOTE: For Limit/RFQ orders we are done here. No need to scale output
-            adjustedFills.push(nativeFill);
+            )[0] as Fill | undefined;
+            // Note: If the order has an adjusted rate of less than or equal to 0 it will be skipped
+            // and nativeFill will be `undefined`
+            if (nativeFill) {
+                // NOTE: For Limit/RFQ orders we are done here. No need to scale output
+                adjustedFills.push({ ...nativeFill, sourcePathId: getSourcePathId(ERC20BridgeSource.Native) });
```

Comment:
Do we wanna do this for all native orders or do we need to separate them into RFQ and Limit?

Reply:
For native orders, I think the sourcePathId doesn't really matter. They always get treated as separate orders when they get serialized.